app/controllers/nonsipp/employercontributions/EmployerContributionsMemberListController.scala

```diff
@@ -63,7 +63,7 @@ class EmployerContributionsMemberListController @Inject()(
 
   def onPageLoad(srn: Srn, page: Int, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) {
     implicit request =>
-      val memberList = request.userAnswers.list(MembersDetailsPages(srn))
+      val memberList = request.userAnswers.map(MembersDetailsPages(srn)).values.toList
```

Comment:
There must be a simplified version too, but IMO this should work:

```
val memberList = request.userAnswers.map(MembersDetailsPages(srn)).toList.sortBy(_._1.toInt).map(_._2)
```

Reply:
Just to summarise our discussion:
Above is just an example, there must be some nicer solution. 
Also this doesn't fix the scenario where there is a gap in the map of members, the employer contribution is still bound to the wrong member.